qemu/tests/qsd_backup_push.py

```diff
@@ -0,0 +1,152 @@
+        qsd_ins = QsdDaemonDev(qsd_name, self.params)
+        return qsd_ins
+
+    def start_qsd(self):
+        self.qsd = self.get_qsd_demon()
+        self.qsd.start_daemon()
+
+    @error_context.context_aware
+    def add_target_data_disks(self):
+        """Hot add target disk to VM with qsd monitor"""
+        error_context.context("Create target disk")
+        for tag in self.params.objects("source_images"):
+            image_params = self.params.object_params(tag)
+            for img in image_params.objects("image_backup_chain"):
+                disk = self.target_disk_define_by_params(self.params, img)
+                disk.hotplug(self.qsd)
+                self.trash.append(disk)
+
+    def do_full_backup(self):
+        extra_options = {"sync": "full", "auto_disable_bitmap": False}
+        backup_utils.blockdev_batch_backup(
+            self.qsd,
+            self.source_images,
+            self.full_backups,
+            self.bitmaps,
+            **extra_options)
+
+    def generate_inc_files(self):
+        for tag in self.params.objects("source_images"):
+            self.generate_data_file(tag)
+
+    def do_increamental_backup(self):
+        extra_options = {'sync': 'incremental', 'auto_disable_bitmap': False}
+        backup_utils.blockdev_batch_backup(
+            self.qsd,
+            self.source_images,
+            self.inc_backups,
+            self.bitmaps,
+            **extra_options)
+
+    def rebase_target_disk(self):
+        self.qsd.stop_daemon()
+        return utils_misc.parallel(self.rebase_targets)
+
+    def prepare_clone_vm(self):
+        self.main_vm.destroy()
+        images = self.params["images"]
+        clone_params = self.main_vm.params.copy()
+        for tag in self.params.objects("source_images"):
+            img_params = self.params.object_params(tag)
+            image_chain = img_params.objects('image_backup_chain')
+            images = images.replace(tag, image_chain[-1])
+            self.params["qsd_images_qsd1"] = image_chain[-1]
+        clone_params["images"] = images
+        clone_vm = self.main_vm.clone(params=clone_params)
+        self.start_qsd()
+        clone_vm.create()
+        clone_vm.verify_alive()
+        self.clone_vm = clone_vm
+
+    def do_test(self):
+        self.do_full_backup()
+        self.generate_inc_files()
+        self.do_increamental_backup()
+        self.main_vm.destroy()
+        self.rebase_target_disk()
+        memory.drop_caches()
+        self.verify_target_disk()
+
+    def verify_target_disk(self):
+        self.prepare_clone_vm()
+        try:
+            self.verify_data_files()
+        finally:
+            self.clone_vm.destroy()
+
+    def prepare_test(self):
+        self.start_qsd()
+        super(QSDBackupTest, self).prepare_test()
+
+    def post_test(self):
+        super(QSDBackupTest, self).post_test()
+        self.qsd.stop_daemon()
+
+
+def run(test, params, env):
+    """
+    increamental backup test via qsd
```

Comment:
suggest add a blank line before "test steps". And suggest keep the same indentation between "incremental" and "test steps".  Thanks.